src/adapted_grid.jl

```diff
@@ -39,8 +38,8 @@ function adapted_grid(f, minmax::Tuple{Real, Real}; max_recursions = 7)
 
     n_tot_refinements = zeros(Int, n_intervals)
 
-    # We only evaluate the function on interior points
-    fs = [NaN; [f(x) for x in xs[2:end-1]]; NaN]
+    # We evaluate the function on the whole interval
+    fs = f.(xs)
```

Comment:
```jl
julia> plot(x->sin(1/x), 0, 1)
ERROR: DomainError with Inf:
sin(x) is only defined for finite x.
Stacktrace:
 [1] sin_domain_error(::Float64) at ./special/trig.jl:28
 [2] sin(::Float64) at ./special/trig.jl:39
 [3] #62 at ./REPL[28]:1 [inlined]
 [4] (::Base.var"#56#57"{Plots.var"#104#105"{Symbol},var"#62#63"})(::Float64) at ./operators.jl:873 (repeats 2 times)
 [5] _broadcast_getindex_evalf at ./broadcast.jl:630 [inlined]
 [6] _broadcast_getindex at ./broadcast.jl:603 [inlined]
 [7] getindex at ./broadcast.jl:563 [inlined]
 [8] copy at ./broadcast.jl:853 [inlined]
 [9] materialize at ./broadcast.jl:819 [inlined]
 [10] #adapted_grid#25(::Int64, ::typeof(adapted_grid), ::Base.var"#56#57"{Base.var"#56#57"{Plots.var"#104#105"{Symbol},var"#62#63"},Plots.var"#106#107"{Symbol}}, ::Tuple{Float64,Float64}) at /home/kc/.julia/packages/PlotUtils/GPrOb/src/adapted_grid.jl:42
 [11] adapted_grid(::Function, ::Tuple{Float64,Float64}) at /home/kc/.julia/packages/PlotUtils/GPrOb/src/adapted_grid.jl:15
```

Reply:
Haha that's the behaviour that was covered under

> that is already handled by the algorithm somehow, at least it doesn't error for me.

@BeastyBlacksmith would you mind taking a look?

Reply:
The easiest is probably just to test `isfinite(f(x))` on the end points and increment as previously if `false`

Reply:
Alternatively, and possibly better, replace non-`finite` values of `x` with `NaN`, which will allow inner asymptotes as well, e.g. `sin(1/x), -1, 1`

Reply:
I don't think testing `isfinite` will work because the problem is that `f` throws `DomainError` exceptions. I think we want to swallow up the `DomainError` exception (anywhere that they happen, not just at the endpoints)

Reply:
Yes I was thinking that too. So wrap the function in try/catch and throw NaN?

Reply:
`g(x) = (try y = f(x); catch; y = NaN; y)` and then use `g`?

Reply:
I don't think all exceptions should be caught.

Reply:
no probably more sensible to just catch `DomainError`?

Reply:
I'll take a look at this

Reply:
Great @BeastyBlacksmith shouldn't be too difficult :-)

Reply:
that was fixed in https://github.com/JuliaPlots/PlotUtils.jl/pull/74

Reply:
:joy: OK I may have a little bit too many things going on at the moment, sorry

Reply:
no worries :)